sessionctx/variable/tidb_vars.go

```diff
@@ -49,6 +49,9 @@ const (
 	// TiDBOptAggPushDown is used to enable/disable the optimizer rule of aggregation push down.
 	TiDBOptAggPushDown = "tidb_opt_agg_push_down"
 
+	// TiDBOptDeriveTopN is used to enable/disable the optimizer rule of deriving topN.
```

Comment:
Should we control this optimizer rule by the blocklist, instead of a new variable?


ref: https://docs.pingcap.com/tidb/stable/blocklist-control-plan#important-optimization-rules

Reply:
I think the use case is different for system variables and blocklist. Blocklist to me seems a configuration which lasts longer. Session/global variables can be used to control a more granular level on  query level as a workaround to force using or not using an optimization.

Reply:
Yes, actually the optimizer rule blocklist is rarely used now, so a system variable seems better here.